scripts/operator-cleanup.sh

```diff
@@ -15,18 +15,34 @@
 # under the License.
 set -x
 
+function rm_operator {
+
+    local NAMESPACE="$1"
+    local OPERATOR="$2"
+
+    if [ -z "$OPERATOR_NAME" ]; then
+        echo "Please set OPERATOR_NAME"; exit 1
+    fi
+
+    local CSV=$(oc get csv --no-headers -o custom-columns=":metadata.name" --ignore-not-found=true | grep $OPERATOR)
+
+    if [ -n "$CSV" ]; then
+      oc delete -n ${NAMESPACE} csv ${CSV} --ignore-not-found=true
+    fi
+    oc delete -n ${NAMESPACE} subscription ${OPERATOR_NAME}-operator --ignore-not-found=true
+    oc delete -n ${NAMESPACE} catalogsource ${OPERATOR_NAME}-operator-index --ignore-not-found=true
+}
+
+
 if [ -z "$NAMESPACE" ]; then
     echo "Please set NAMESPACE"; exit 1
 fi
 
-if [ -z "$OPERATOR_NAME" ]; then
-    echo "Please set OPERATOR_NAME"; exit 1
-fi
 
-CSV=$(oc get csv --no-headers -o custom-columns=":metadata.name" --ignore-not-found=true | grep $OPERATOR_NAME)
+# List all the operators in the ${NAMESPACE}
+OPERATORS=$(oc get subscriptions.operators.coreos.com -n ${NAMESPACE} -o jsonpath='{range .items[*]}{.spec.name}{"\n"}{end}')
 
-if [ -n "$CSV" ]; then
-    oc delete -n ${NAMESPACE} csv ${CSV} --ignore-not-found=true
-fi
-oc delete -n ${NAMESPACE} subscription ${OPERATOR_NAME}-operator --ignore-not-found=true
-oc delete -n ${NAMESPACE} catalogsource ${OPERATOR_NAME}-operator-index --ignore-not-found=true
+# For each operator delete the associated resources (csv/subs/catalogsource)
+for operator in $OPERATORS; do
+    rm_operator $NAMESPACE $operator
+done
```

Comment:
This seems to remove the the CSV and `Subscriptions` of all operators I have deployed in the namespace, regardless of which `make <operator>_cleanup` I run.  i.e. I had both `MariaDB` and `Keystone` (via `make mariadb && make keystone`), but then running `make keystone_cleanup` seemed to also remove `MariaDB`.

Reply:
ooh right, because we're using the same script and filter by <operator>.
I'll do another iteration and remove all the operators only if `make openstack_cleanup` is called.

Reply:
Done